### PR TITLE
[ryml] Add new port

### PR DESCRIPTION
**Pull request by vvandrounik**
**Describe the pull request**

- What does your PR fix?
N/A
- Which triplets are supported/not supported? Have you updated the CI baseline?
Tested (locally): x64/x86-linux, x64/x86-windows (static linkage only). 
No.
- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
Yes.

### CODE REVIEW

File: ports/ryml/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,66 @@
+include(vcpkg_common_functions)
```
Comment (LilyWangL) [inline ports/ryml/portfile.cmake:1]: `include(vcpkg_common_functions)` is deprecated. Please remove it.

**Review by LilyWangL — CHANGES_REQUESTED** [review]

File: ports/ryml/portfile.cmake (hunk 1 above)
Comment (vvandrounik) [reply ports/ryml/portfile.cmake]: Done 3ac4f46

**Comment by vvandrounik** [comment]
Done

File: ports/ryml/portfile.cmake (hunk 1 above)
Comment (vvandrounik) [reply ports/ryml/portfile.cmake]: Done 3ac4f46

**Comment by LilyWangL** [comment]
This port depends on c4core, so waiting for PR #10791 merge.

**Comment by vvandrounik** [comment]
Sorry, I have two left hands. I accidentally did merged master branch into ryml and pushed, so I did used force push to overwrite this. 😢

**Comment by vvandrounik** [comment]
I'm should ignore otl regression and update CI baseline?

**Comment by LilyWangL** [comment]
> I'm should ignore otl regression and update CI baseline?



> I'm should ignore otl regression and update CI baseline?

This is the CI error, please wait it fixed by other PRs.

**Comment by JackBoosY** [comment]
The otl regressions(#10922) are not related with this PR.

**Review by LilyWangL — APPROVED** [review]

**Comment by ras0219-msft** [comment]
LGTM, thanks for the PR!

**Pull request merged**